torch/csrc/autograd/record_function.h

```diff
@@ -107,6 +107,20 @@ struct TORCH_API RecordFunction {
 
   void end();
 
+  // Saves the thread_id that this RecordFunction was created with. This is
+  // needed so that we can access Events created by the original thread in a
+  // different thread, since they are thread-local. This should be used to call
+  // RecordFunction::end() in a different thread.
+  void setThreadId();
+
+  // Retrieves the thread_id that this RecordFunction was created with. Useful
+  // if we need to access Events created by the original thread in a different
+  // thread. The threadId_ should only be set in cases where RecordFunction::end
```

Comment:
nit: Can you specify what "set" means here? ` threadId_ != 0`?

Reply:
Clarified this to mean that the user should call `setThreadId()`.